src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,486 @@
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
+
+  const hasText = !!(isDefined(input) && input.length > 0);
+  const hasAnimatedLabel = variant === 'outlined';
+
+  const placeholderTextColor = getPlaceholderTextColor({
+    theme,
+    disabled,
+    isPlaceholderVisible,
+    hasAnimatedLabel,
+  });
+
+  // Derive accessory conditions based on props
+  const { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon } = getAccessoryConfig({
+    errorText,
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  if (labelPosition === 'left' && variant === 'outlined') {
+    // Outlined Text Input has only a top label
+    throw Error('Cannot have a left label on an outlined Text Input');
+  }
+
+  if (!isEmpty(prefix) && !isEmpty(iconLeft)) {
+    throw Error('Cannot have prefix and left icon together');
+  }
+
+  if (!isEmpty(suffix) && !isEmpty(iconRight)) {
+    throw Error('Cannot have suffix and right icon together');
+  }
+
+  const onFocus = useCallback(() => {
+    setIsFocused(true);
+
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, []);
+
+  const onBlurText = useCallback(
+    (event) => {
+      setIsFocused(false);
+      setIsPlaceholderVisible(false);
+      const text = event.target.value;
+      setInput(text);
+      if (onBlur) {
+        onBlur(text);
+      }
+    },
+    [setIsFocused, setIsPlaceholderVisible, setInput, onBlur],
+  );
+
+  const onChangeText = useCallback(
+    (event) => {
+      const text = event.target.value;
+      if (text.length > maxLength) {
+        return;
+      }
+      setInput(text);
+      if (onChange) {
+        onChange(text);
+      }
+    },
+    [maxLength, onChange, setInput],
+  );
+
+  const onSelectText = () => {
+    return theme.colors.primary[980];
+  };
+
+  const onKeyPress = (e) => {
+    if (type === 'number') {
+      const charCode = typeof e.which === 'number' ? e.which : e.keyCode;
+      const char = String.fromCharCode(charCode);
+      const isAllowed = /[0-9]/g.test(char) || char === '-' || char === '.';
+
+      if (!isAllowed) {
+        e.preventDefault();
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (isDefined(value)) {
+      setInput(value);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (inputRef.current && containerRef.current) {
+      const containerRect = containerRef.current.getBoundingClientRect();
+      const inputRect = inputRef.current.getBoundingClientRect();
+      const initialLeftPosition = inputRect.x - containerRect.x;
+      const finalTopPosition = inputRef.current.offsetHeight;
+
+      setLayoutDimensions({
+        initialLeftPosition,
+        finalTopPosition,
+      });
+    }
+  }, [inputRef, containerRef]);
```

Comment:
Add `setLayoutDimensions`

Reply:
saw that setState hooks can be omitted as they are stable across re-renders. will be adding anyways.